### ISSUE TITLE
run postgres_extensible query across multiple databases

### ISSUE THREAD

**Issue by phemmer** [influxdata/telegraf#3219]
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal:
The postgres_extensible plugin should support a way of running the same query across multiple databases. The configuration should allow running across all databases, or a filtered list.

### Current behavior:
No such feature

### Desired behavior:
Such a feature

### Use case: [Why is this important (helps with prioritizing requests)]
Postgres provides numerous statistics though what it calls ["system catalogs"](https://www.postgresql.org/docs/9.3/static/catalogs.html), which are really tables within each database. Thus to use them you have to connect to that database. Instead of having to know which databases are present, and then having to hard-code those databases in the telegraf config on each server, it would be nice if we could push out the same telegraf config to all servers.

**Comment by danielnelson** [comment]
What about the [databases](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/postgresql_extensible/postgresql_extensible.go#L61-L63) option?  It sounds like it does something similar, though maybe we need better filtering.

**Comment by phemmer** [comment]
No, the `databases` option is nothing like what being requested. The `databases` option simply tacks `IN ('foo','bar','baz')` on to the end of the query, which I personally don't see any use for.

**Comment by phemmer** [comment]
I'm really thinking the current `databases` option should either be removed, or replaced with this feature request. I can see no benefit to it. The 2 use cases in the documentation are invalid:

> New queries can be added, if the withdbname is set to true and there is no
  databases defined in the 'databases field', the sql query is ended by a 'is
  not null' in order to make the query succeed.

The user can add `is not null` to their query. There's no reason to make it more complicated by setting a config parameter (`withdbname=true`) in which all it does is add the `is not null` to the end of the query for you.

> the plugin will add a 'IN (dbname list)' clause if the withdbname is set to true

All this does is take the values of the `databases` parameter, join them into a quoted list, and append it to the query. Again there's no reason to make this complicated when the user can add this to the query directly.

**Comment by danielnelson** [comment]
I don't understand why someone would use it either, if anyone is using this option currently can you explain how it is helpful?

Even though it is using a good name, we might want to call the new feature `dbnames` and leave databases alone outside of deprecating it.

**Comment by SanderDR** [comment]
I would still like this feature to come through. When querying stats for all tables in all databases with `select * from pg_stat_all_tables`, I would need to define the plugin once for every database in my database server. It would be handy if I could just define the query and it would run against all discovered databases in my database server.

**Comment by satyanash** [comment]
The `withdbname` flag is incredibly non-intuitive. It will append either an `IS NOT NULL` or `IN (db list)` clause depending on the `databases` field in the `inputs.postgresql_extensible` configuration.

However, the utility of these clauses is not explained at all. What kind of queries will be able to take these and ensure it works correctly for postgres is also not documented.
